src/spec.c

```diff
@@ -1518,12 +1500,6 @@ StrongRef IndexSpec_LoadUnsafeEx(RedisModuleCtx *ctx, IndexLoadOptions *options)
   IndexSpec_IncreasCounter(sp);
 
   if (!RS_IsMock && (sp->flags & Index_Temporary) && !(options->flags & INDEXSPEC_LOAD_NOTIMERUPDATE)) {
-    if (sp->isTimerSet) {
-      WeakRef old_timer_ref;
-      if (RedisModule_StopTimer(RSDummyContext, sp->timerId, (void **)&old_timer_ref) == REDISMODULE_OK) {
-        WeakRef_Release(old_timer_ref);
-      }
-    }
     IndexSpec_SetTimeoutTimer(sp, StrongRef_Demote(spec_ref));
```

Comment:
Why `RedisModule_StopTimer` was replace with `IndexSpec_SetTimeoutTimer`?

Reply:
it wasn't replaced. we previously checked and canceled the timer before calling this function, but this function also deals with the timer internally (code duplication). I removed the external call because the internal one is used by other flows that call `IndexSpec_SetTimeoutTimer`.